core/src/test/scala/unit/kafka/admin/ConfigCommandTest.scala

```diff
@@ -472,6 +512,134 @@ class ConfigCommandTest extends ZooKeeperTestHarness with Logging {
     EasyMock.reset(alterResult, describeResult)
   }
 
+  @Test
+  def shouldNotAlterNonQuotaIpConfigsUsingBootstrapServer(): Unit = {
+    // when using --bootstrap-server, it should be illegal to alter anything that is not a connection quota
+    // for ip entities
+    val node = new Node(1, "localhost", 9092)
+    val mockAdminClient = new MockAdminClient(util.Collections.singletonList(node), node)
+
+    def verifyCommand(entityType: String, alterOpts: String*): Unit = {
+      val opts = new ConfigCommandOptions(Array("--bootstrap-server", "localhost:9092",
+        "--entity-type", entityType, "--entity-name", "admin",
+        "--alter") ++ alterOpts)
+      val e = intercept[IllegalArgumentException] {
+        ConfigCommand.alterConfig(mockAdminClient, opts)
+      }
+      assertTrue(s"Unexpected exception: $e", e.getMessage.contains("some_config"))
+    }
+
+    verifyCommand("ips", "--add-config", "connection_creation_rate=10000,some_config=10")
```

Comment:
super tiny nit: `verifyCommand` name makes the code a bit confusing since the command is expected to fail. What about `verifyCommandFails` or `verifyFails`?

Reply:
I do agree. It took me a while to grasp it. Moreover, the very same `verifyCommand` is defined in other places. I wonder if we could consolidate them into one. What do you think?